### PR TITLE
feat: traverse symlinks in the workspace

### PR DESCRIPTION
**Pull request by hubertmis**
A likeC4 workspace can contain symbolic links to documents shared between multiple workspaces. The WorspaceManager is enhanced to traverse through symbolic links to correctly include documents linked by symbolic links.

### CODE REVIEW

**Comment by hubertmis** [comment]
@davydkov , do you have any hints why playground can't find the node:fs module?

File: packages/language-server/src/shared/WorkspaceManager.ts [hunk 1]
```diff
@@ -27,6 +28,54 @@ export class LikeC4WorkspaceManager extends DefaultWorkspaceManager {
...
+
+  protected async followDirectoryEntry(
+    entryPath: URI,
+    workspaceFolder: WorkspaceFolder,
+    folderPath: URI,
+    fileExtensions: string[],
+    collector: (document: LangiumDocument) => void
+  ): Promise<void> {
+    const uri = entryPath
+    const stat = await fs.promises.lstat(uri.fsPath)
```
Comment (davydkov) [inline packages/language-server/src/shared/WorkspaceManager.ts]: This is redundant call, as  `includeEntry` does `lstatSync`

**Review by davydkov — CHANGES_REQUESTED** [review]
Hey @hubertmis 

I see what you're trying to solve, but you did not take into account that language server (and vscode extension) can be run as webworker in browser (and playground does it), and thus `node:fs` is not available there.

Check:
- `startLanguageServer` in node runtime, [index.ts](https://github.com/likec4/likec4/blob/ee04db787862985c1b723cefea6cd6f1db4857ec/packages/language-server/src/index.ts#L2) langium provided with `NodeFileSystem`
- `startLanguageServer` in browser [browser.ts](https://github.com/likec4/likec4/blob/main/packages/language-server/src/browser.ts)

I suggest you provide node runtime only implementation via dependency-injection (like `WorkspaceCrawler.ts`)
Later, we may want to configure it, turn on/off resolving symlinks, add `includes/excludes` (as in recent [discussion #1174](https://github.com/likec4/likec4/discussions/1174))
[`fdir`](https://github.com/thecodrr/fdir) seems a perfect fit.

Or consider to override [`NodeFileSystem`](https://github.com/eclipse-langium/langium/blob/main/packages/langium/src/node/node-file-system-provider.ts) and traverse symlinks there?

And don't forget about [CLI](https://github.com/likec4/likec4/blob/ee04db787862985c1b723cefea6cd6f1db4857ec/packages/likec4/src/language/module.ts#L142) (also initialised with `NodeFileSystem`)

**Comment by hubertmis** [comment]
Thanks for the hints! I'll have a look at the modules you suggested

**Comment by hubertmis** [comment]
I don't know what "dependency-injection" is and I could not find the `WorkspaceCrawler.ts` example. However, traversing symlinks in overridden `NodeFileSystem` seemed clear enough for me. It seems to work fine!

**Review by davydkov — APPROVED** [review]
>  I could not find the WorkspaceCrawler.ts example

Sorry, I was not clear, I mean create new file and name it like `WorkspaceCrawler.ts`

I merge this PR, and I move `LikeC4FileSystem` out of `index.ts`

**Pull request merged**